tidy3d/components/data/data_array.py

```diff
@@ -650,6 +653,47 @@ class HeatDataArray(DataArray):
     _dims = "T"
 
 
+class EMEScalarFieldDataArray(AbstractSpatialDataArray):
+    """Spatial distribution of a mode in frequency-domain as a function of mode index
+    and port index.
+
+    Example
+    -------
+    >>> x = [1,2]
+    >>> y = [2,3,4]
+    >>> z = [3,4,5,6]
+    >>> f = [2e14, 3e14]
+    >>> mode_index = np.arange(5)
+    >>> port_index = [0, 1]
+    >>> coords = dict(x=x, y=y, z=z, f=f, mode_index=mode_index, port_index=port_index)
+    >>> fd = ScalarModeFieldDataArray((1+1j) * np.random.random((2,3,4,2,5,2)), coords=coords)
+    """
+
+    __slots__ = ()
+    _dims = ("x", "y", "z", "f", "mode_index", "port_index")
+
+
+class EMESMatrixDataArray(DataArray):
```

Comment:
This is different than the smatrix plugin data array, which allows more than two ports. This, together with the EMESMatrixDataset which has S11, S12, S21, S22, is better for the two-port case. It doesn't have the problem of possible different num_modes at the two ports. Also, we can't import smatrix plugin from components.